axelor-production/src/main/java/com/axelor/apps/production/service/MrpServiceProductionImpl.java

```diff
@@ -443,60 +465,76 @@ protected void createProposalMrpLine(
       throws AxelorException {
 
     Company company = mrp.getStockLocation().getCompany();
-    BillOfMaterial defaultBillOfMaterial = billOfMaterialService.getDefaultBOM(product, company);
+
+    Optional<MrpLine> originalMrpLineOpt =
+        mrpLineOriginList.stream().findAny().map(MrpLineOrigin::getMrpLine);
+
+    BillOfMaterial billOfMaterial = null;
+    if (originalMrpLineOpt.map(MrpLine::getBillOfMaterial).isPresent()) {
```

Comment:
There is something weird with this bloc : 
```java
  if (originalMrpLineOpt.map(MrpLine::getBillOfMaterial).isPresent()) {
      billOfMaterial =
          billOfMaterialMrpLineService
              .getEligibleBillOfMaterialOfProductInMrpLine(originalMrpLineOpt.orElse(null), product)
              .orElse(null);
    }
    if (billOfMaterial == null) {
      billOfMaterial = billOfMaterialService.getDefaultBOM(product, company);
    }
```

You don't need to do this check `originalMrpLineOpt.map(MrpLine::getBillOfMaterial).isPresent()`, because `billOfMaterialMrpLineService
              .getEligibleBillOfMaterialOfProductInMrpLine` will already do the check and in this specific case  will always return an Optional.of(mrpLine.getBillOfMaterial) (since its present).
What is weird is that I don't understand why you call `getEligibleBillOfMaterialOfProductInMrpLine` in this case since this part of the method will never be called 

```java
    if (product == null) {
      return Optional.empty();
    }

    Optional<SaleOrderLine> saleOrderLineOpt =
        MrpLineTool.getOriginSaleOrderLineInMrpLineOrigin(mrpLine);
    Optional<BillOfMaterial> billOfMaterialOpt =
        saleOrderLineOpt.map(SaleOrderLine::getBillOfMaterial);

    if (billOfMaterialOpt.isPresent()
        && product.equals(billOfMaterialOpt.map(BillOfMaterial::getProduct).orElse(null))) {
      return billOfMaterialOpt;
    }
    return Optional.empty();
``` 
If this is intended then the method `getElgilibleBillOfMaterialOfProductInMrpLine` has a part of its code that will never be reached and therefore useless (since it seems it is the only place where the method is called)

In all cases, the bloc can be replaced with 
```java
    billOfMaterial = mrpLineOriginList
            .stream()
            .findAny()
            .map(MrpLineOrigin::getMrpLine)
            .flatMap(mrpLine -> billOfMaterialMrpLineService.getEligibleBillOfMaterialOfProductInMrpLine(mrpLine, product))
            .orElse(billOfMaterialService.getDefaultBOM(product, company));
``` 
(Need to remove the unreached part of the code)

However if you wish to keep the unreached code because you actually call the method somewhere elese then this can be replaced instead with

```java
    billOfMaterial = mrpLineOriginList
            .stream()
            .findAny()
            .map(MrpLineOrigin::getMrpLine)
            .map(MrpLine::getBillOfMaterial)
            .orElse(billOfMaterialService.getDefaultBOM(product, company));
```